test/show.jl

```diff
@@ -2768,3 +2768,9 @@ let topmi = ccall(:jl_new_method_instance_uninit, Ref{Core.MethodInstance}, ());
     topmi.def = Main
     @test contains(repr(topmi), "Toplevel MethodInstance")
 end
+
+@testset "show(<do-block expr>) no trailing whitespace" begin
+    do_expr1 = :(foo() do; bar(); end)
+    @test !contains(sprint(show, do_expr1), " \n")
+    @test sprint(show, do_expr1) == "foo() do\n    bar()\nend"
```

Comment:
```suggestion
```
I would just remove this failing test. I don't think it's worth dealing with line number nodes here, nor is it particularly beneficial to add such a strict (and therefore fragile) test about exact printing.

The bug this PR fixes is that a space precedes a newline and the test above checks for just that and fails on master and passes on the PR, demonstrating the bug and that it is fixed.

Reply:
thanks :)